Mathlib/Computability/ContextFreeGrammar.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2023 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Computability.Language
+
+/-!
+# Context-Free Grammar
+
+This file contains the definition of a Context-Free Grammar (CFG), which is a grammar that has a
+single nonterminal symbol on the left-hand side of each rule.
+Derivation by a grammar is inherently nondeterministic.
+-/
+
+/-- The type of symbols is the disjoint union of terminals `T` and nonterminals `N`. -/
+inductive Symbol (T : Type*) (N : Type*)
```

Comment:
This shouldn't be in the global namespace, as other languages use `Symbol` in different ways; can you put it (and maybe all the new types) inside the `ContextFreeGrammar` namespace?